propagation/propagation.go

```diff
@@ -23,97 +45,159 @@ import (
 //  context=${contextBlob} - contextBlob is a base64 encoded json object.
 //
 // ex: X-Honeycomb-Trace: 1;trace_id=weofijwoeifj,parent_id=owefjoweifj,context=SGVsbG8gV29ybGQ=
+type HoneycombContext struct{}
 
-const (
-	TracePropagationHTTPHeader = "X-Honeycomb-Trace"
-	TracePropagationVersion    = 1
-)
-
-type Propagation struct {
-	TraceID      string
-	ParentID     string
-	Dataset      string
-	TraceContext map[string]interface{}
-}
-
-type PropagationError struct {
-	message      string
-	wrappedError error
+// HoneycombHTTPPropagator returns an HTTPPropagator that understands Honeycomb
+// trace context headers.
+func HoneycombHTTPPropagator() trace.HTTPPropagator {
```

Comment:
I vote for either not exporting `HoneycombContext` **or** returning `HoneycombContext` here. Don't return the interface type unless you're trying to hide the concrete type.

Reply:
I agree. Opted to return `HoneycombContext` (and `AmazonContext` below).